rclcpp/include/rclcpp/callback_group.hpp

```diff
@@ -136,12 +144,14 @@ class CallbackGroup
   CallbackGroupType type_;
   // Mutex to protect the subsequent vectors of pointers.
   mutable std::mutex mutex_;
+  std::atomic_bool associated_with_executor_;
   std::vector<rclcpp::SubscriptionBase::WeakPtr> subscription_ptrs_;
   std::vector<rclcpp::TimerBase::WeakPtr> timer_ptrs_;
   std::vector<rclcpp::ServiceBase::WeakPtr> service_ptrs_;
   std::vector<rclcpp::ClientBase::WeakPtr> client_ptrs_;
   std::vector<rclcpp::Waitable::WeakPtr> waitable_ptrs_;
   std::atomic_bool can_be_taken_from_;
+  std::atomic_bool allow_executor_to_add_;
```

Comment:
I propose to consider changing the type to `const bool` as it is set a construction time only. This might make testing this flag slightly faster.

Reply:
I agree. After considering between different options and testing, const bool is what we need because the `allow_executor_to_add_` should not change in the life of the callback group.